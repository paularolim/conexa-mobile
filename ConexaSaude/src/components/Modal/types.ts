export interface ModalProps {
  title: string;
  description: string;
  primaryButtonLabel: string;
  secondaryButtonLabel: string;
  visible: boolean;
  dismissModal: () => void;
  confirm: () => void;
  cancel: () => void;
}
