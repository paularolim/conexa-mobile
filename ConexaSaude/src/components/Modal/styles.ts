import { View } from 'react-native';
import styled from 'styled-components/native';

export const Overlay = styled(View)`
  background-color: #00000066;
  flex: 1;
  justify-content: flex-end;
`;

export const Container = styled(View)`
  padding: 24px;
  border-top-left-radius: 12px;
  border-top-right-radius: 12px;
  gap: 24px;
  background-color: ${({ theme }) => theme.colors.white};
`;

export const Section = styled(View)`
  gap: 4px;
`;
