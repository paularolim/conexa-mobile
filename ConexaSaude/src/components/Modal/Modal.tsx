import { Modal as RNModal, TouchableWithoutFeedback } from 'react-native';

import { Button } from '@components/Button';
import { Text } from '@components/Text';

import { Container, Overlay, Section } from './styles';
import { ModalProps } from './types';

export function Modal({
  title,
  description,
  primaryButtonLabel,
  secondaryButtonLabel,
  visible,
  dismissModal,
  cancel,
  confirm,
}: ModalProps) {
  return (
    <RNModal visible={visible} transparent>
      <TouchableWithoutFeedback onPress={dismissModal}>
        <Overlay>
          <Container>
            <Section>
              <Text style={{ fontSize: 22, fontWeight: 'bold' }}>{title}</Text>
              <Text style={{ fontSize: 16 }}>{description}</Text>
            </Section>

            <Section>
              <Button onPress={confirm}>{primaryButtonLabel}</Button>
              <Button type="ghost" onPress={cancel}>
                {secondaryButtonLabel}
              </Button>
            </Section>
          </Container>
        </Overlay>
      </TouchableWithoutFeedback>
    </RNModal>
  );
}
