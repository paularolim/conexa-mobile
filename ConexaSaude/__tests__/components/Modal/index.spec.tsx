import renderer from 'react-test-renderer';

import { Modal } from '@components/Modal';

import { Wrapper } from '../../Wrapper';

describe('Modal component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Modal
            title="Modal title"
            description="Modal description"
            primaryButtonLabel="Primary Button Label"
            secondaryButtonLabel="Secondary Button Label"
            visible={false}
            dismissModal={jest.fn()}
            confirm={jest.fn()}
            cancel={jest.fn()}
          />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
