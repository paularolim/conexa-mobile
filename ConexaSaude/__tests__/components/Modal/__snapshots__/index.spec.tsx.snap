// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render default component correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;
